extensions/documents/homie meta extension.md

```diff
@@ -0,0 +1,121 @@
+# Meta
+
+Version: **<!--VERSION-->1.0.0<!--VERSION-->**
+Date: **<!--DATE-->10. Jul 2019<!--DATE-->**
+Authors: **<!--AUTHORS-->[EPNW](https://epnw.eu)<!--AUTHORS-->**
+License: **<!--LICENSE-->[CCA 4.0](https://homieiot.github.io/license)<!--LICENSE-->**
+
+## Abstract
+This extension defines how to add metadata and tags to devices, nodes and properties.
+
+Tags are simple annotations that ever device, node oder property can have.
+Metadata on the other hand are more complex. They allow the definition of multiple *mainkeys* and *mainvalues* for each device, node or property.
+Each main-key-value-pair may have nested sub-key-value-pairs.
+Having metadata might be useful for Homie controllers.
+A concrete usecase is the openHAB controller, which relies on metadata and tags to make voice assistants like Amazons Alexa or the Google Assistant able to control it.
+With this extension, properties are able to advertise how they want to be treated.
+
+## Extension Identifier
+The ID of this extension is `eu.epnw.meta`.
+
+## Homie Version
+This extension supports Homie `3.0.1` and `4.0`.
+
+## Extension Datatypes
+This extension defines no new datatypes.
+
+## Extension Attributes
+
+### Extension Attributes for any kind of Items
+
+The following attributes are all **optional** and may be added to any kind of item (devices, nodes or properties).
```

Comment:
I stated in the extension convention that every new attribute has to be retained, so per definition, this attributes are retained, too. But since you mention this here, I feel like we should discuss if requiring every extension attribute to be retained is a good idea.